api/gin_engine_setup_test.go

```diff
@@ -0,0 +1,70 @@
+package api_test
+
+import (
+	"context"
+	"net/http"
+	"net/http/httptest"
+
+	. "github.com/onsi/ginkgo"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/fabric8-services/fabric8-wit/api"
+	"github.com/fabric8-services/fabric8-wit/gormapplication"
+	"github.com/fabric8-services/fabric8-wit/gormsupport/cleaner"
+	"github.com/fabric8-services/fabric8-wit/gormtestsupport"
+)
+
+// Execute submits the request and returns the response recording fo subseauent verifications
+func Execute(s gormtestsupport.GinkgoTestSuite, r *http.Request) *httptest.ResponseRecorder {
```

Comment:
we should avoid duplicating this utility method. What about moving this test in the `api/handler` pkg instead ? you could rename the `gin_engine_setup_test.go` file into `redirect_test.go` and verify that the call to `GET /api/foobar` is redirected, as you've already done in this PR. Also, could you verify that `POST`, `PATCH` and `DELETE` requests are also correctly handled ?

Reply:
Yes

Reply:
Fixed!